.github/workflows/static_analysis.yml

```diff
@@ -0,0 +1,29 @@
+name: Static analysis
+
+on:
+  pull_request:
+    branches:
+      - "2.0"
+
+jobs:
+  pre-commit-checks:
+    name: Pre-commit checks
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.10
```

Comment:
Pls always use quotation marks when using floats where strings are expected. Without them, this could be interpreted as `3.1`.